qt/scientific_interfaces/Indirect/test/IndirectFitDataTest.h

```diff
@@ -0,0 +1,447 @@
+    auto const combinedData = data2.combine(data1);
+
+    TS_ASSERT_EQUALS(combinedData.getExcludeRegion(0), "1.0,2.0,5.0,6.0");
+    TS_ASSERT_EQUALS(combinedData.getExcludeRegion(1), "2.0,6.0");
+  }
+
+  void
+  test_that_the_spectra_pair_of_two_datasets_are_combined_correctly_when_spectra_do_not_overlap() {
+    auto data1 = getIndirectFitData(10, 3);
+    auto data2 = getIndirectFitData(10, 3);
+
+    data1.setSpectra(std::make_pair(0u, 4u));
+    data2.setSpectra(std::make_pair(5u, 9u));
+    auto const combinedData = data2.combine(data1);
+    Spectra const spec(std::make_pair(0u, 9u));
+
+    TS_ASSERT(
+        boost::apply_visitor(AreSpectraEqual(), combinedData.spectra(), spec));
+  }
+
+  void
+  test_that_the_spectra_pair_of_two_datasets_are_combined_correctly_when_spectra_are_discontinuous() {
+    auto data1 = getIndirectFitData(10, 3);
+    auto data2 = getIndirectFitData(10, 3);
+
+    data1.setSpectra(std::make_pair(0u, 4u));
+    data2.setSpectra(std::make_pair(8u, 9u));
+    auto const combinedData = data2.combine(data1);
+    Spectra const spec(DiscontinuousSpectra<std::size_t>("0-4,8-9"));
+
+    TS_ASSERT(
+        boost::apply_visitor(AreSpectraEqual(), combinedData.spectra(), spec));
+  }
+
+  void
+  test_that_the_spectra_pair_of_two_datasets_are_combined_correctly_when_spectra_overlap() {
+    auto data1 = getIndirectFitData(10, 3);
+    auto data2 = getIndirectFitData(10, 3);
+
+    data1.setSpectra(std::make_pair(0u, 8u));
+    data2.setSpectra(std::make_pair(4u, 9u));
+    auto const combinedData = data2.combine(data1);
+    Spectra const spec(DiscontinuousSpectra<std::size_t>("0-9"));
+
+    TS_ASSERT(
+        boost::apply_visitor(AreSpectraEqual(), combinedData.spectra(), spec));
+  }
+
+  void
+  test_that_the_DiscontinuousSpectra_of_two_datasets_are_combined_correctly_when_spectra_do_not_overlap() {
+    auto data1 = getIndirectFitData(10, 3);
+    auto data2 = getIndirectFitData(10, 3);
+
+    data1.setSpectra(DiscontinuousSpectra<std::size_t>("0-4"));
+    data2.setSpectra(DiscontinuousSpectra<std::size_t>("5-9"));
+    auto const combinedData = data2.combine(data1);
+    Spectra const spec(DiscontinuousSpectra<std::size_t>("0-9"));
+
+    TS_ASSERT(
+        boost::apply_visitor(AreSpectraEqual(), combinedData.spectra(), spec));
+  }
+
+  void
+  test_that_the_DiscontinuousSpectra_of_two_datasets_are_combined_correctly_when_spectra_overlap() {
+    auto data1 = getIndirectFitData(10, 3);
+    auto data2 = getIndirectFitData(10, 3);
+
+    data1.setSpectra(DiscontinuousSpectra<std::size_t>("0-7"));
+    data2.setSpectra(DiscontinuousSpectra<std::size_t>("2-9"));
+    auto const combinedData = data2.combine(data1);
+    Spectra const spec(DiscontinuousSpectra<std::size_t>("0-9"));
+
+    TS_ASSERT(
+        boost::apply_visitor(AreSpectraEqual(), combinedData.spectra(), spec));
+  }
+
+  void
+  test_that_a_SpectraPair_and_DiscontinuousSpectra_dataset_are_combined_correctly_when_spectra_do_not_overlap() {
```

Comment:
is a `SpectraPair` a defined object like `DiscontinuousSpectra`? If not maybe use `Spectra_pair` to avoid confusion.